service/history/ndc/resetter.go

```diff
@@ -153,6 +153,10 @@ func (r *resetterImpl) resetWorkflow(
 	}
 	rebuildMutableState.AddHistorySize(rebuiltHistorySize)
 
+	if err := rebuildMutableState.RefreshExpirationTimeoutTask(ctx); err != nil {
```

Comment:
just curious why this is not be done in the state rebuilder

Reply:
This is because of how reset is done in active workflow. `r.stateRebuilder.Rebuild` is called from  `func (r 
*workflowResetterImpl) replayResetWorkflow`. Which is called from `prepareResetWorkflow`.  So adding that to `replay' doesn't feel like the right place, thus I add it to `prepareResetWorkflow`